File: 回溯/78.子集.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def backTrack(combine, start):
            res.append(combine)  # append修改元素组没有返回值
            for i in range(start, len(nums)):
                backTrack(combine + [nums[i]], i + 1)
        backTrack([], 0)
        return res 

File: 回溯/78.子集.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function (nums) {
  const res = [];
  const combine = [];
  function backtrack(start) {
    res.push([...combine]);
    for (let i = start; i < nums.length; i++) {
      combine.push(nums[i]);
      backtrack(i + 1);
      combine.pop();
    }
  }
  backtrack(0);
  return res;
};
